packages/hardhat/test/ChainLance.ts

```diff
@@ -13,15 +13,240 @@ describe("ChainLance", function () {
+    await txCreateCompletedProject.wait();
+
+    const bidId4 = "bidid4";
+    await chainLance.connect(worker1).bidProject(completedProjectId, bidId4, completedProjectPrice, 1);
+
+    const txAcceptBid4 = await chainLance
+      .connect(employer3)
+      .acceptBid(completedProjectId, bidId4, { value: completedProjectPrice });
+    await txAcceptBid4.wait();
+
+    const txSubmitWork4 = await chainLance.connect(worker1).submitWork(completedProjectId);
+    await txSubmitWork4.wait();
+
+    const txAcceptWork = await chainLance.connect(employer3).acceptWork(completedProjectId);
+    await txAcceptWork.wait();
+    ///////////////////////////////////////////////////////////////////
+    // Work with state: Canceled
+    const canceledProjectId = "id6";
+    const canceledProjectPrice = 313131;
+    const txCreateCanceledProject = await chainLance
+      .connect(employer2)
+      .createProject(canceledProjectId, canceledProjectPrice, 100);
+    await txCreateCanceledProject.wait();
+
+    const bidId5 = "bidid5";
+    await chainLance.connect(worker1).bidProject(canceledProjectId, bidId5, canceledProjectPrice, 1);
+
+    const bidId6 = "bidid6";
+    await chainLance.connect(worker2).bidProject(canceledProjectId, bidId6, canceledProjectPrice, 1);
+
+    const bidId7 = "bidid7";
+    await chainLance.connect(worker3).bidProject(canceledProjectId, bidId7, canceledProjectPrice, 1);
+
+    await chainLance.connect(employer2).cancelProject(canceledProjectId);
+  });
+
+  it("Should list projects in different states", async function () {
+    const openProjects = await chainLance.listProjectsWithState(0);
+    expect(openProjects).to.include("id42");
+
+    const inWorkProjects = await chainLance.listProjectsWithState(1);
+    expect(inWorkProjects).to.include("id3");
+
+    const inReviewProjects = await chainLance.listProjectsWithState(2);
+    expect(inReviewProjects).to.include("id4");
+
+    const completedProjects = await chainLance.listProjectsWithState(3);
+    expect(completedProjects).to.include("id5");
+
+    const canceledProjects = await chainLance.listProjectsWithState(4);
+    expect(canceledProjects).to.include("id6");
+  });
+});
+
+describe("ChainLance: check access rights", function () {
+  // We define a fixture to reuse the same setup in every test.
+
+  let chainLance: ChainLance;
+  before(async () => {
+    // const [owner] = await ethers.getSigners();
+    const chainLanceFactory = await ethers.getContractFactory("ChainLance");
+    chainLance = (await chainLanceFactory.deploy()) as ChainLance;
+    await chainLance.waitForDeployment();
+    // TODO: create some projects in different states and several bids on them to check different
+    // they can be used in all "it" sections
+  });
+
+  it("Should allow to change project state to employer only", async function () {
+    // TODO
+  });
+
+  it("Should allow to change work state to worker only", async function () {
+    // TODO
```

Comment:
implement this case

Reply:
Can't use vals from "before" in "It"

Reply:
this vals should be declared as consts or var outside the `before` section